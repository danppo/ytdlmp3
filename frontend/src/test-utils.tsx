import * as React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider, theme, ColorModeScript } from "@chakra-ui/react";

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <>
    <ColorModeScript />
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  </>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options });

export { customRender as render };
