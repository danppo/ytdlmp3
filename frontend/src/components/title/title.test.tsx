import React from 'react';
import { screen } from '@testing-library/react';
import { render } from '../../test-utils';
import Title from './title';

// beforeEach(() => render(<Title />));

const titleValue = 'test title';

describe('renders components:', () => {
  test('self', () => {
    render(<Title value={titleValue} size='m' />);
    const currentElement = screen.getByTestId('title');
    expect(currentElement).toBeInTheDocument();
    expect(currentElement).toHaveTextContent(titleValue);
  });

  describe('test size param:', () => {
    test('medium ', () => {
      render(<Title value={titleValue} size='m' />);
      const currentElement = screen.getByTestId('title');
      expect(currentElement).toMatchSnapshot();
    });
    test('3xl ', () => {
      render(<Title value={titleValue} size='3xl' />);
      const currentElement = screen.getByTestId('title');
      expect(currentElement).toMatchSnapshot();
    });
  });
  describe('test bold param:', () => {
    test('normal ', () => {
      render(<Title value={titleValue} size='m' />);
      const currentElement = screen.getByTestId('title');
      expect(currentElement).toMatchSnapshot();
    });
    test('bold ', () => {
      render(<Title value={titleValue} size='m' bold />);
      const currentElement = screen.getByTestId('title');
      expect(currentElement).toMatchSnapshot();
    });
  });
});
