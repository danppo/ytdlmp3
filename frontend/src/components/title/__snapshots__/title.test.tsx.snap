// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components: test bold param: bold  1`] = `
<b
  class="chakra-text css-1s6bgx9"
  data-testid="title"
>
  test title
</b>
`;

exports[`renders components: test bold param: normal  1`] = `
<p
  class="chakra-text css-1s6bgx9"
  data-testid="title"
>
  test title
</p>
`;

exports[`renders components: test size param: 3xl  1`] = `
<p
  class="chakra-text css-1jijfcn"
  data-testid="title"
>
  test title
</p>
`;

exports[`renders components: test size param: medium  1`] = `
<p
  class="chakra-text css-1s6bgx9"
  data-testid="title"
>
  test title
</p>
`;
